File: ts-sdk/src/sdk/models/shared/userv2.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class UserV2Properties extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=name" })
  name: string;

  @SpeakeasyMetadata({ data: "json, name=value" })
  value: string;
}

export enum UserV2StatusEnum {
    Active = "Active",
    Pending = "Pending",
    Deactivated = "Deactivated",
    Deleted = "Deleted"
}


export class UserV2 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=created_at" })
  createdAt?: string;

  @SpeakeasyMetadata({ data: "json, name=department" })
  department?: string;

  @SpeakeasyMetadata({ data: "json, name=display_name" })
  displayName?: string;

  @SpeakeasyMetadata({ data: "json, name=draft_email" })
  draftEmail?: string;

  @SpeakeasyMetadata({ data: "json, name=email" })
  email?: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=image_uri" })
  imageUri?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=is_signature_enabled" })
  isSignatureEnabled?: boolean;

  @SpeakeasyMetadata({ data: "json, name=mfa_enabled" })
  mfaEnabled?: boolean;

  @SpeakeasyMetadata({ data: "json, name=organization_id" })
  organizationId?: string;

  @SpeakeasyMetadata({ data: "json, name=phone" })
  phone?: string;

  @SpeakeasyMetadata({ data: "json, name=phone_verified" })
  phoneVerified?: boolean;

  @SpeakeasyMetadata({ data: "json, name=preferred_language" })
  preferredLanguage?: string;

  @SpeakeasyMetadata({ data: "json, name=properties", elemType: UserV2Properties })
  properties?: UserV2Properties[];

  @SpeakeasyMetadata({ data: "json, name=signature" })
  signature?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: UserV2StatusEnum;

  @SpeakeasyMetadata({ data: "json, name=token" })
  token?: string;
}

File: user-api/go-sdk/pkg/models/shared/userv2.go
package shared

type UserV2Properties struct {
	Name  string `json:"name"`
	Value string `json:"value"`
}

type UserV2StatusEnum string

const (
	UserV2StatusEnumActive      UserV2StatusEnum = "Active"
	UserV2StatusEnumPending     UserV2StatusEnum = "Pending"
	UserV2StatusEnumDeactivated UserV2StatusEnum = "Deactivated"
	UserV2StatusEnumDeleted     UserV2StatusEnum = "Deleted"
)

type UserV2 struct {
	CreatedAt          *string                `json:"created_at,omitempty"`
	Department         *string                `json:"department,omitempty"`
	DisplayName        *string                `json:"display_name,omitempty"`
	DraftEmail         *string                `json:"draft_email,omitempty"`
	Email              *string                `json:"email,omitempty"`
	ID                 *string                `json:"id,omitempty"`
	ImageURI           map[string]interface{} `json:"image_uri,omitempty"`
	IsSignatureEnabled *bool                  `json:"is_signature_enabled,omitempty"`
	MfaEnabled         *bool                  `json:"mfa_enabled,omitempty"`
	OrganizationID     *string                `json:"organization_id,omitempty"`
	Phone              *string                `json:"phone,omitempty"`
	PhoneVerified      *bool                  `json:"phone_verified,omitempty"`
	PreferredLanguage  *string                `json:"preferred_language,omitempty"`
	Properties         []UserV2Properties     `json:"properties,omitempty"`
	Signature          *string                `json:"signature,omitempty"`
	Status             *UserV2StatusEnum      `json:"status,omitempty"`
	Token              *string                `json:"token,omitempty"`
}
